### ISSUE TITLE
cli: introduce `-f` when adding grants, instead of 'not found checks' being the default behavior

### ISSUE THREAD

**Issue by kimskimchi** [infrahq/infra#1787]
## Is your feature request related to a problem? Please describe.

Currently, user can add a grant to a user, role, and resource that infra does not know about yet. 
Ideally this should be checked, since the current experience is more of "grant access, and then test to see if it actually works as intended", where as the goal hopefully would be "grant access, and be sure that it works if the command is successful" 

## Describe the solution you'd like

1. User should exist. If it doesn't command should fail, unless `--force` or `-f` is set. 2. 
2. Role should be a subset of valid ones, based on the type of the destination. It should fail if the role is not found for that type of destination. (eg. infra has a set of roles, and it should fail if it is not one of them)3. 
3. Resource/destination should exist. Infra should be able to check this somehow, and if it is not connected to infra (aka infra has no way of granting user to that resource because it doesn't know about it), then it should fail. 
4. All of the above can be overridden, and access can be granted, if user provides the boolean `--force` flag. It will succeed, add the grant into the system, while displaying a warning message. 

## Environment Details

```
$ infra version
```

```
$ kubectl version
```

**Comment by kimskimchi** [comment]
assigning to Patrick for any additional comments I might have missed

**Comment by pdevine** [comment]
Yep, this is correct. Unfortunately there are some caveats here:
1. We don't currently have a way to check for roles. This should be done with something like `GET /v1/destinations/%s/roles` (exact API call TBD)
2. We don't have a way of checking whether the destination resource is valid (i.e. if it's a namespace or some other resource) We'll need to explore this more to see how we can do it.
3. Given that we don't have SCIM, we can't know whether a user exists inside of a provider (with the exception of local users).

**Comment by pdevine** [comment]
Also, we should return exit code 1 if we're unsuccessful, and 0 if the command succeeds. Using `-f` would make the command succeed regardless of whether we can validate the role/destination/user.
